src/pods/generate/GenerateUtil.ts

```diff
@@ -13,7 +13,7 @@ import type { ResourcesGenerator } from './ResourcesGenerator';
  */
 export async function addGeneratedResources(identifier: ResourceIdentifier, settings: NodeJS.Dict<string>,
   generator: ResourcesGenerator, store: ResourceStore): Promise<number> {
-  const resources = generator.generate(identifier, settings);
+  const resources = generator.generate(identifier, { podBaseUrl: identifier.path, ...settings });
```

Comment:
@joachimvh I'm not happy with this one. I spent quite some time puzzling around with `VariableHandler`, only to found out that they're not passed to the templates 🤦‍♀️

What _is_ passed, I found quite worrying though:

```json
{
  email: 'ruben@verborgh.org',
  password: 'j04ch1mr0cks',
  confirmPassword: 'j04ch1mr0cks',
  createWebId: 'on',
  register: 'on',
  webId: 'http://localhost:3000/ruben/profile/card#me',
  createPod: 'on',
  podName: 'ruben',
  submit: '',
  oidcIssuer: 'http://localhost:3000/'
}
```

1. Probably shouldn't be sending password around
2. This is arbitrary JSON from registration leaking all the way into pod generation; we likely want to decouple those because this reliance seems quite fragile.

Reply:
The `VariableHandler` is something specifically for the dynamic pods to validate the values being injected into the configs 😅 

>   1. Probably shouldn't be sending password around.

I thought about this when making the registration handler. In the end I didn't remove them because the only time this would be an issue is if you have templates storing the password. So that would imply that you just can't trust the server in general. Someone would have to write new templates that contain the password fields and set up a server with those. So the only risk is the person setting up the server themselves.

>   2. This is arbitrary JSON from registration leaking all the way into pod generation; we likely want to decouple those because this reliance seems quite fragile.

>   3. Worse, it's even a dependency on the the exact HTML structure (`'on'` rather than boolean, `submit`).

The data actually gets cleaned in the registration handler: https://github.com/solid/community-server/blob/fe8d579c72bb5b146e638647644252f05e4bd353/src/identity/interaction/email-password/handler/RegistrationHandler.ts#L49-L62

But indeed, that cleaned data is not what we send to the templates, there the raw data is used. The reason behind that was to support someone setting up a custom registration page that accepts more fields than we require to fill in the templates. E.g., full name, address, etc. To support that we need to send along all the fields that we get as input.

One thing we could potentially do to clean this up is to update the raw object with all the values from the interface above, but keep the raw values for the other ones.

Reply:
Thanks, tracking in https://github.com/solid/community-server/issues/845